Change the signature `foo()` to `foo(void)` to avoid warnings

```
* checking whether package 'string2path' can be installed ... WARNING
Found the following significant warnings:
  rust/api.h:7:1: warning: function declaration isn't a prototype [-Wstrict-prototypes]
  init.c:69:6: warning: function declaration isn't a prototype [-Wstrict-prototypes]
```

--- a/R-package/src/init.c
+++ b/R-package/src/init.c
@@ -32,12 +32,12 @@ SEXP handle_result(SEXP res_) {
     return (SEXP)res;
 }
 
-SEXP safe_stop__impl() {
+SEXP safe_stop__impl(void) {
     SEXP res = safe_stop();
     return handle_result(res);
 }
 
-SEXP raise_error__impl() {
+SEXP raise_error__impl(void) {
     SEXP res = raise_error();
     return handle_result(res);
 }
@@ -87,7 +87,7 @@ SEXP print_list__impl(SEXP x) {
     return handle_result(res);
 }
 
-SEXP Person_new__impl() {
+SEXP Person_new__impl(void) {
     SEXP res = Person_new();
     return handle_result(res);
 }
@@ -102,7 +102,7 @@ SEXP Person_name__impl(SEXP self__) {
     return handle_result(res);
 }
 
-SEXP Person_associated_function__impl() {
+SEXP Person_associated_function__impl(void) {
     SEXP res = Person_associated_function();
     return handle_result(res);
 }
@@ -127,22 +127,22 @@ SEXP scalar_input_string__impl(SEXP x) {
     return handle_result(res);
 }
 
-SEXP scalar_output_int__impl() {
+SEXP scalar_output_int__impl(void) {
     SEXP res = scalar_output_int();
     return handle_result(res);
 }
 
-SEXP scalar_output_real__impl() {
+SEXP scalar_output_real__impl(void) {
     SEXP res = scalar_output_real();
     return handle_result(res);
 }
 
-SEXP scalar_output_logical__impl() {
+SEXP scalar_output_logical__impl(void) {
     SEXP res = scalar_output_logical();
     return handle_result(res);
 }
 
-SEXP scalar_output_string__impl() {
+SEXP scalar_output_string__impl(void) {
     SEXP res = scalar_output_string();
     return handle_result(res);
 }

--- a/R-package/src/rust/api.h
+++ b/R-package/src/rust/api.h
@@ -1,5 +1,5 @@
-SEXP safe_stop();
-SEXP raise_error();
+SEXP safe_stop(void);
+SEXP raise_error(void);
 SEXP to_upper(SEXP x);
 SEXP add_suffix(SEXP x, SEXP y);
 SEXP times_two_int(SEXP x);
@@ -13,15 +13,15 @@ SEXP scalar_input_int(SEXP x);
 SEXP scalar_input_real(SEXP x);
 SEXP scalar_input_logical(SEXP x);
 SEXP scalar_input_string(SEXP x);
-SEXP scalar_output_int();
-SEXP scalar_output_real();
-SEXP scalar_output_logical();
-SEXP scalar_output_string();
+SEXP scalar_output_int(void);
+SEXP scalar_output_real(void);
+SEXP scalar_output_logical(void);
+SEXP scalar_output_string(void);
 SEXP sum_int(SEXP x);
 SEXP sum_real(SEXP x);
 
 // methods and associated functions for Person
-SEXP Person_new();
+SEXP Person_new(void);
 SEXP Person_set_name(SEXP self__, SEXP name);
 SEXP Person_name(SEXP self__);
-SEXP Person_associated_function();
+SEXP Person_associated_function(void);

--- a/savvy-bindgen/src/gen/c.rs
+++ b/savvy-bindgen/src/gen/c.rs
@@ -23,14 +23,18 @@ impl SavvyFn {
     /// Generate C function signature
     fn to_c_function_for_header(&self) -> String {
         let fn_name = self.fn_name_outer();
-        let args = self
-            .get_c_args()
-            .iter()
-            .map(|(pat, ty)| format!("{ty} {pat}"))
-            .collect::<Vec<String>>()
-            .join(", ");
+        let args = self.get_c_args();
 
-        format!("SEXP {fn_name}({args});")
+        let args_sig = if args.is_empty() {
+            "void".to_string()
+        } else {
+            args.iter()
+                .map(|(pat, ty)| format!("{ty} {pat}"))
+                .collect::<Vec<String>>()
+                .join(", ")
+        };
+
+        format!("SEXP {fn_name}({args_sig});")
     }
 
     /// Generate C function implementation
@@ -38,17 +42,23 @@ impl SavvyFn {
         let fn_name = self.fn_name_outer();
         let args = self.get_c_args();
 
-        let args_sig = args
-            .iter()
-            .map(|(pat, ty)| format!("{ty} {pat}"))
-            .collect::<Vec<String>>()
-            .join(", ");
+        let (args_sig, args_call) = if args.is_empty() {
+            ("void".to_string(), "".to_string())
+        } else {
+            let args_sig = args
+                .iter()
+                .map(|(pat, ty)| format!("{ty} {pat}"))
+                .collect::<Vec<String>>()
+                .join(", ");
 
-        let args_call = args
-            .iter()
-            .map(|(pat, _)| pat.as_str())
-            .collect::<Vec<&str>>()
-            .join(", ");
+            let args_call = args
+                .iter()
+                .map(|(pat, _)| pat.as_str())
+                .collect::<Vec<&str>>()
+                .join(", ");
+
+            (args_sig, args_call)
+        };
 
         format!(
             "SEXP {fn_name}__impl({args_sig}) {{

